### PR TITLE
#192 roles view -> list/create

### PR DESCRIPTION
**Pull request by Vatavuk**
For #192 , created implementations for list/create and left one puzzle for further development.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#231](https://github.com/timelesslounge/timelessis/issues/231) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#231](https://github.com/timelesslounge/timelessis/issues/231) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/87e18179-2957-49ba-ac9a-cec38b870163); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: migrations/versions/2019-02-12T194215.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    sa.Column('updated_on', sa.DateTime(), nullable=False),
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('name', sa.String(), nullable=False),
+    sa.Column('code', sa.String(), nullable=False),
+    sa.Column('address', sa.String(), nullable=True),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('code'),
+    sa.UniqueConstraint('name')
+    )
+    op.create_table('customers',
```
Comment (vladarefiev) [inline migrations/versions/2019-02-12T194215.py]: @vatavuk why do we have this one? Is it mistake?

File: migrations/versions/2019-02-12T194215.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    sa.Column('synchronized_on', sa.DateTime(), nullable=True),
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('first_name', sa.String(), nullable=False),
+    sa.Column('last_name', sa.String(), nullable=False),
+    sa.Column('phone_number', sa.String(), nullable=False),
+    sa.Column('created_on', sa.DateTime(), nullable=False),
+    sa.Column('updated_on', sa.DateTime(), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table('reservation_settings',
```
Comment (vladarefiev) [inline migrations/versions/2019-02-12T194215.py]: @Vatavuk this one has already created too

File: tests/test_roles.py [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
     assert client.get("/roles/").status_code == HTTPStatus.OK
 
 
 def test_create(client):
-    assert client.get("/roles/create").status_code == HTTPStatus.OK
+    response = client.post(url_for("role.create"), data={
+        "name": "owner"
+    })
+    assert response.location.endswith(url_for('role.list_roles'))
+    assert Role.query.count() == 1
```
Comment (vladarefiev) [inline tests/test_roles.py]: @Vatavuk May be let's check exactly this row, not checking only count? WDYT?

**Comment by vladarefiev** [comment]
@Vatavuk looks great, but a couple of comments

File: migrations/versions/2019-02-12T194215.py (hunk 1 above)
Comment (Vatavuk) [reply migrations/versions/2019-02-12T194215.py]: @vladarefiev I will remove migrations from PR, we don't need them, I've added them by mistake

File: tests/test_roles.py (hunk 3 above)
Comment (Vatavuk) [reply tests/test_roles.py]: @vladarefiev 👍

**Comment by Vatavuk** [comment]
@vladarefiev fixed

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 37 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#231](https://github.com/timelesslounge/timelessis/issues/231) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**